components/datetime/src/date.rs

```diff
@@ -1,18 +1,34 @@
+use std::convert::{TryFrom, TryInto};
+use std::ops::{Add, Sub};
+use std::str::FromStr;
+
+#[derive(Debug)]
+pub enum DateTimeError {
+    Parse(std::num::ParseIntError),
+    Overflow(&'static str),
```

Comment:
Suggestion: make this `Overflow { field: &'static str, max: usize }` and generate the string by implementing a Display trait.  This is what I've done in FixedDecimal and DataProvider.  Avoiding hard-coded strings deep inside application logic (centralizing them into the Display traits) might make it easier to support internationalizing the error messages down the road.

Reply:
Done!